### ISSUE TITLE
Script as a Service

### ISSUE THREAD

**Issue by coutodiogo** [anetrusov/ZabbixDBA#75]
Hi Alexander,

In your scenario, has the perl script (perl /opt/zdba/zdba.pl /opt/zdba/conf/config.pl) been configured to start together with the system as a service? After running, if the terminal session is terminated, the script is also finished.

**Comment by netrusov** [comment]
hi @coutodiogo 

there's init script that you can use to enable automatic startup of zdba (see `/init.d/zdba` in repo), but don't forget to review it before implementing it into your system
also you can run script manually in background by adding `&` at the end (standard shell functionality)

**Comment by coutodiogo** [comment]
Hi @anetrusov 

I configure the script as a service, but when i execut "service zdba start" the error above is shown in /opt/zdba/log/zdba.err :

"Can't locate forks.pm in @INC (you may need to install the forks module) (@INC contains: /etc/perl /usr/local/lib/perl/5.18.2 /usr/local/share/perl/5.18.2 /usr/lib/perl5 /usr/share/perl5 /usr/lib/perl/5.18 /usr/share/perl/5.18 /usr/local/lib/site_perl .) at /opt/zdba/ZabbixDBA.pl line 6.
BEGIN failed--compilation aborted at /opt/zdba/ZabbixDBA.pl line 6."

If I run the script manually, the monitor works. I had this same problem previously, running the script manually, and you suggested me to reinstall the forks module (I commented on your youtube video with doubts).

What do you think it could be? Would it be appropriate to reinstall the forks?  The manual script works, only the parameterized via service that does not.

**Comment by netrusov** [comment]
did you configure `RUN_AS` variable properly? if it is currently running as `root` it doesn't see `perlbrew` and can't find installed modules.
also, you can add `source /path/to/perl5/perlbrew/etc/bashrc` at the beginning of `/etc/init.d/zdba` and it should work.

**Comment by coutodiogo** [comment]
I follow your instructions and didn't work. I run the command "perl /opt/zdba/zdba.pl /opt/zdba/conf/config.pl" as root and works.

**Comment by netrusov** [comment]
but it works when you run it manually, so the issue is with environment variables.
in order perl to work properly it requires `PERL5LIB` environment variable which contains paths to libraries.
just try to add `source /path/to/perl5/perlbrew/etc/bashrc` at the beginning of the file and run it again.

**Comment by coutodiogo** [comment]
[Doc1.docx](https://github.com/anetrusov/ZabbixDBA/files/1318948/Doc1.docx)

**Comment by coutodiogo** [comment]
@anetrusov this attachment contains some prints. I have already tried to add all the paths below in the file:

`source /path/to/perl5/perlbrew/etc/bashrc`
`source ~/perl5/perlbrew/etc/bashrc`
`source /root/perl5/perlbrew/etc/bashrc`

**Comment by netrusov** [comment]
just like I said - it didn't load paths. just look at the list of directories aftter `@INC contains:` - it doesn't contain `perlbrew` paths
also, `/path/to/perl5/perlbrew/etc/bashrc` is not valid, I just didn't know where you've installed `perlbrew`
as I see in your comment on YT, it should be `source /root/perl5/perlbrew/etc/bashrc`
and send me output of `perlbrew list`, please.

**Comment by coutodiogo** [comment]
The output is  `perl-5.26.0`

**Comment by coutodiogo** [comment]
@anetrusov I tried using the command `source /root/perl5/perlbrew/etc/bashrc` but it did not work

**Comment by lpossamai** [comment]
I am having the same issue. Also added `source /root/perl5/perlbrew/etc/bashrc` to the init.d file and that did not work.

I can see, however, the init.d file loads the `/root/perl5/perlbrew/etc/bashrc`:
```
echo $PATH
/root/perl5/perlbrew/bin:/root/perl5/perlbrew/perls/perl-5.26.0/bin:
```

Error is:
```
Can't locate forks.pm in @INC (@INC contains: /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5 .) at /opt/zdba/ZabbixDBA.pl line 6.
BEGIN failed--compilation aborted at /opt/zdba/ZabbixDBA.pl line 6.
```

**Comment by lpossamai** [comment]
UPDATE:

I've fixed it by doing the following (Red Hat 6.8):

1. vim /etc/init.d/zdba

#Add this line:
`source /root/perl5/perlbrew/etc/bashrc`

2. vim ~/.bash_profile

#Add this line:
`source /root/perl5/perlbrew/etc/bashrc`

3. service zdba start

**Comment by netrusov** [comment]
@lpossamai thanks for update!
